Groups might not be published in ascending order (#793)
moq-wg/moq-transport: draft-ietf-moq-transport.md

the latest group(s).  The publisher then selectively transmits objects based on
their group membership.  Groups can contain any number of objects.

### Group Ordering

Within a track, the original publisher SHOULD publish Group IDs which increase
with time. In some cases, Groups will be produced in increasing order, but sent
to subscribers in a different order, for example when the subscription's Group
Order is Descending.  Due to network reordering and the partial reliability
features of MoQT, Groups can always be received out of order.

As a result, subscribers cannot infer the existence of a Group until an object in

> the Group is received. This can create gaps in the MoQ cache that MUST be filled
> by doing a Fetch upstream if a Fetch is received that requests those Groups.

Reviewer: ```suggestion
the Group is received. This can create gaps in a cache that can be filled
by doing a Fetch upstream, if necessary.
```

Subscribers to tracks which do not follow this requirement SHOULD NOT
use range filters which span multiple groups in FETCH or SUBSCRIBE. SUBSCRIBE and
FETCH delivery use Group Order, so a FETCH cannot deliver Groups out of order
and a subscription could have unexpected delivery order if Group IDs do not increase

> with time.

Reviewer: Maybe this should be merged with the paragraph below, the first sentence redundant now, and the second sentence doesn't make sense.

the latest group(s).  The publisher then selectively transmits objects based on
their group membership.  Groups can contain any number of objects.

### Group Ordering

Within a track, the original publisher SHOULD publish Group IDs which increase
with time. There are cases when even if the Groups are produced in increasing order,
they might not be sent to the MoQ network or various Subscribers in the network
in that order. One example is if the original publisher is publishing in descending
Group Order.

As a result, Subscribers will not know the status of a Group within a SUBSCRIBE's
Group range, even if there is no Delivery Timeout for the Subscription, until

> either the Group is received or a 'Group Does Not Exist' status is received
> for the Group.  This can create gaps in the MoQ cache that MUST be filled by

Author: ```suggestion
either the Group is received or a 'Group Does Not Exist' status is received.
This can create gaps in the MoQ cache that MUST be filled by
```

doing a Fetch upstream if a Fetch is received that requests those Groups.

> Subscribers to tracks which do not follow this requirement SHOULD NOT

Author: I'm not sure we need this paragraph, or possibly it deserves a rewrite?

Reviewer: Maybe something like this:

Applications that cannot produce groups with IDs that increase with time are limited to the subset of MoQT that does not numerically compare group IDs.  Subscribers to these tracks SHOULD NOT use range filters which span multiple groups in FETCH or SUBSCRIBE.

Author: I took your suggestion with a minor rewrite.
